packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -28,11 +29,13 @@ export default class GoogleCalendarService implements Calendar {
   private integrationName = "";
   private auth: { getToken: () => Promise<MyGoogleAuth> };
   private log: typeof logger;
+  private credential: CredentialWithAppName;
 
-  constructor(credential: CredentialPayload) {
+  constructor(credential: CredentialWithAppName) {
     this.integrationName = "google_calendar";
     this.auth = this.googleAuth(credential);
     this.log = logger.getChildLogger({ prefix: [`[[lib] ${this.integrationName}`] });
+    this.credential = credential;
   }
 
   private googleAuth = (credential: CredentialPayload) => {
```

Comment:
This credential is no longer needed as googleAuth can access this.credential

Reply:
We needed the email of the owner of the credential to not include their email in the list of team members to prevent duplicate events. `googleAuth` doesn't contain this. To get around it I've made `credentialUserEmail` a parameter in the create event method.

Reply:
I do think this is a very destructive refactor. Feels like we're patching instead of solving for long term.